corehq/apps/case_search/utils.py

```diff
@@ -65,13 +67,22 @@ def _assemble_optional_search_params(self):
         self._add_daterange_queries()
         self._add_case_property_queries()
 
+    def _validate_param_value(self, key, value):
+        if isinstance(value, list):
+            raise CaseFilterError(
+                _("Multiple values for this param is not supported"),
+                key
+            )
+
     def _add_xpath_query(self):
         query = self.criteria.pop(CASE_SEARCH_XPATH_QUERY_KEY, None)
+        self._validate_param_value(CASE_SEARCH_XPATH_QUERY_KEY, query)
```

Comment:
Do you think it would be clearer if you moved all of this into a separate method and called it at the top?
Something like:

```python
def _validate_multiple_parameter_values(self):
    disallowed_multiple_value_parameters = [
        CASE_SEARCH_BLACKLISTED_OWNER_ID_KEY,
        'owner_id',
        CASE_SEARCH_XPATH_QUERY_KEY,
    ]

    for key, val in self.criteria.items():
        if not isinstance(val, list):
            continue
        if key in disallowed_multiple_value_parameters or '/' in key or val.startswith('__range__') :
            raise CaseFilterError(...)
```
Currently it isn't super clear which things should accept multiple values and which ones shouldn't.

Reply:
Sure, updated.